test/runner/configuration/include_test.dart

```diff
@@ -0,0 +1,148 @@
+      d.file('dart_test_base.yaml', r'''
+        filename: "test_*.dart"
+      '''),
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: ../dart_test_base.yaml
+          concurrency: 3
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    var config = new Configuration.load(path);
+    expect(config.filename.pattern, 'test_*.dart');
+    expect(config.concurrency, 3);
+  });
+
+  test('should merge fields with a base configuration', () async {
+    await d.dir('repo', [
+      d.file('dart_test_base.yaml', r'''
+        tags:
+          hello:
+      '''),
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: ../dart_test_base.yaml
+          tags:
+            world:
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    var config = new Configuration.load(path);
+    expect(config.knownTags, unorderedEquals(['hello', 'world']));
+  });
+
+  test('should allow an included file to include a file', () async {
+    await d.dir('repo', [
+      d.file('dart_test_base_base.yaml', r'''
+        tags:
+          tag:
+      '''),
+      d.file('dart_test_base.yaml', r'''
+        include: dart_test_base_base.yaml
+        filename: "test_*.dart"
+      '''),
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: ../dart_test_base.yaml
+          concurrency: 3
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    var config = new Configuration.load(path);
+    expect(config.knownTags, ['tag']);
+    expect(config.filename.pattern, 'test_*.dart');
+    expect(config.concurrency, 3);
+  });
+
+  test('should handle a non-string include field value gracefully', () async {
+    await d.dir('repo', [
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: 3 # Oops!
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    expect(() => new Configuration.load(path), throwsFormatException);
+  });
+
+  test('should handle a missing include gracefully', () async {
+    await d.dir('repo', [
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: other_test.yaml # Oops!
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    expect(() => new Configuration.load(path), throwsFormatException);
+  });
+
+  test('should not allow an include field in a test config context', () async {
+    await d.dir('repo', [
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          tags:
+            foo:
+              include: ../dart_test.yaml
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    expect(() => new Configuration.load(path), throwsFormatException);
```

Comment:
I ran this locally as a sanity check, and it looks like this is getting a "cannot open file" error rather than a "this field isn't allowed here" error. Maybe these throw checks should be a little more specific about what error message is being produced...

Reply:
Done. PTAL.